app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -578,6 +578,34 @@ public function getEntities($args = array())
             ) : $results;
     }
 
+
+    public function getEntitiesLight($args = array()){
+    	$limit = 20000; //lines
+    	$results = array();
+    	$offset = 0;
+    	$result = $this->requestEntitiesLight($args, $offset, $limit);
+    	$results = array_merge($results, $result);
+    	while (count($result) === $limit){
+    		$offset += $limit;
+    		$result = $this->requestEntitiesLight($args, $offset+1, $limit);
+    		$results = array_merge($results, $result);
+    	}
+    	return $results;
+    }
```

Comment:
@escopecz From what I understand, using the `$model->getEntities()` method collected so much unused data from the database (such as contact fields) that it would fill the memory much more quickly than the present method which limits the amount of requested information.

I think the most viable *real* solution would be to make a generalized *export* function which would be able to **stream** the database data to a csv file, freeing the memory as it goes along.